src/ast/data_type.rs

```diff
@@ -25,10 +25,21 @@ use serde::{Deserialize, Serialize};
 #[cfg(feature = "visitor")]
 use sqlparser_derive::{Visit, VisitMut};
 
-use crate::ast::{display_comma_separated, ObjectName, StructField, UnionField};
+use crate::ast::{display_comma_separated, ObjectName, StructField, UnionField, Value};
 
 use super::{value::escape_single_quote_string, ColumnDef};
 
+#[derive(Debug, Clone, PartialEq, PartialOrd, Eq, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[cfg_attr(feature = "visitor", derive(Visit, VisitMut))]
+pub enum EnumValue {
+    String(String),
+    /// ClickHouse allows to specify an integer value for each enum value.
+    ///
+    /// [clickhouse](https://clickhouse.com/docs/en/sql-reference/data-types/enum)
+    Pair(String, Value),
+}
```

Comment:
```suggestion
pub enum EnumMember {
    Name(String),
    /// ClickHouse allows to specify an integer value for each enum value.
    ///
    /// [clickhouse](https://clickhouse.com/docs/en/sql-reference/data-types/enum)
    NamedValue(String, Value),
}
```
Would something like this be reasonable? thinking member is convention and the variant would be a bit more descriptive

Reply:
Yes, That's a good suggestion.